### PR TITLE
Start using Centos 8 as base

### PR DESCRIPTION
**Pull request by elfosardo**

### CODE REVIEW

**Review by dtantsur — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
-    yum update -y && \
-    yum install -y openstack-ironic-python-agent lshw smartmontools iproute python-hardware mdadm biosdevname ipmitool && \
-    yum clean all
+RUN dnf install -y python3 python3-requests && \
+    curl https://raw.githubusercontent.com/openstack/tripleo-repos/master/tripleo_repos/main.py | python3 - -b train current && \
+    dnf install -y epel-release 'dnf-command(config-manager)' && \
+    dnf config-manager --set-enabled PowerTools && \
+    dnf update -y && \
+    dnf install -y openstack-ironic-python-agent lshw smartmontools \
+      iproute python3-hardware mdadm biosdevname ipmitool && \
```
Comment (dtantsur) [inline Dockerfile]: We need `python3-hardware-detect`, not `python3-hardware`. The latter pulls unnecessary bits (for now).

**Review by dtantsur — APPROVED** [review]
/assign @juliakreger 

One comment inline. It was a problem before, so not blocking.

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,16 @@
-FROM docker.io/centos:centos7
+FROM docker.io/centos:centos8
 
-RUN yum install -y python-requests && \
-    curl https://raw.githubusercontent.com/openstack/tripleo-repos/master/tripleo_repos/main.py | python - -b train current-tripleo && \
-    yum update -y && \
-    yum install -y openstack-ironic-python-agent lshw smartmontools iproute python-hardware mdadm biosdevname ipmitool && \
-    yum clean all
+RUN dnf install -y python3 python3-requests && \
+    curl https://raw.githubusercontent.com/openstack/tripleo-repos/master/tripleo_repos/main.py | python3 - -b train current && \
```
Comment (hardys) [inline Dockerfile:4]: Note that current isn't the same as current-tripleo - we're now consuming packages which haven't passed tripleo-ci, and previously that has provided some confidence that Ironic is likely to be working - IMO we should consider switching back to current-tripleo when that is available.

**Pull request merged**